### PR TITLE
introduce the "meta" field in Tunable class

### PR DESCRIPTION
**Pull request by motus**
We'll need it for many things, but right now the plan is to store the units and scalers for some tunable values and use it for MySQL, e.g., to produce config values like `innodb_buffer_pool_size = 128M`

### CODE REVIEW

File: mlos_bench/mlos_bench/tunables/tunable.py [hunk 1]
```diff
@@ -63,6 +64,7 @@ def __init__(self, name: str, config: TunableDict):
         self._description = config.get("description")
         self._default = config["default"]
         self._values = config.get("values")
+        self._meta: Dict[str, Any] = config.get("meta", {})
```
Comment (bpkroth) [inline mlos_bench/mlos_bench/tunables/tunable.py:67]:
```suggestion
        self._meta: Dict[str, TunableValue] = config.get("meta", {})
```

Do you forsee a case where it should be more than this?

File: mlos_bench/mlos_bench/tunables/tunable.py [hunk 2]
```diff
@@ -376,3 +378,11 @@ def categories(self) -> List[Optional[str]]:
...
         assert self._values is not None
         return self._values
+
+    @property
+    def meta(self) -> Dict[str, Any]:
+        """
+        Get the tunable's metadata. This is a free-form dictionary that can be used to
+        store any additional information about the tunable (e.g., the unit information).
+        """
+        return self._meta
```
Comment (bpkroth) [inline mlos_bench/mlos_bench/tunables/tunable.py:388]: What about during the conversion to ConfigSpace?

Comment (motus) [reply mlos_bench/mlos_bench/tunables/tunable.py:388]: good point! we should add it to the `meta` field there. Let's do it in the next PR, along with the unit tests

File: mlos_bench/mlos_bench/tunables/tunable.py (hunk 1 above)
Comment (motus) [reply mlos_bench/mlos_bench/tunables/tunable.py:67]: actually, yes - e.g., something along the line of:
```json
"multiplier_suffix": {
    1024: "K",
    1048576: "M",
    1073741824: "G"
}
```
or, say,
```json
"constraints": {
    "less_than": ["innodb_io_capacity_max"]
}
```
etc.

File: mlos_bench/mlos_bench/tunables/tunable.py (hunk 2 above)
Comment (bpkroth) [reply mlos_bench/mlos_bench/tunables/tunable.py:388]: Please make Issues to track the future PRs so we don't lose track of them.  Thanks!

**Review by bpkroth — APPROVED** [review]

File: mlos_bench/mlos_bench/tunables/tunable.py (hunk 2 above)
Comment (motus) [reply mlos_bench/mlos_bench/tunables/tunable.py:388]: Created issue #419

**Pull request merged**